### ISSUE TITLE
Allow composition

### ISSUE THREAD

**Issue by eimfach** [Vahera/godot-orchestrator#441]
### Description

Allow composition of multiple scripts attached to one node. 

### Implementation ideas

Make a new orchestration script as a composition of multiple ones. All the graphs and components would be placed inside this new script, inside a colored comment like block with the source scripts name, to visually order the components according to their main script.
Optionally you can decide wether to just copy and paste or link the components (so they will change if the parent changes); I also think there need to be nodes defined for each "source block" individually that are for "exposing" internal values of the block, if the components are linked. So that if the intervals change nothing needs to changed in the api of the linked components (because the internals can not be directly wired only the exposed nodes) unless there is refactoring of types going on.

Long story short: Basically Modules.

**Comment by Naros** [comment]
So, I think there are many ways you can work with composition.

You can do this in UE through a combination of macro and function libraries.  We're already working on the former, as we think those provide the best prefab reuse features, allowing users to define custom nodes that can be used across multiple Orchestration scripts much the same way you may reuse a `.gd` script in multiple other script sources.

However, I am concerned about this ideology of attaching multiple scripts to the same node.  I'm concerned that most would use this poorly, leading to poor programming habits and blurring the separation of concerns principle.

Let's put Orchestrator aside for just a moment.   

Today, if you wanted to explore composition in Godot, you would have an instanced scene representing some prefab, perhaps a door, a player, etc.  Let's say this door object has multiple concerns like collision and player interactions. You would likely have two child nodes where each would have its own script, and that child node handles that specific concern.  If these two siblings need to interact with one another, you would generally use a blackboard in the parent or provide a way for the child nodes to gain a reference to a sibling component, much like how ECS/Component systems work in Unity.

What exactly would this module concept offer, perhaps a more concrete example would help my understand why such a nested architecture would be needed within the scripting subsystem rather than using the scene tree for this.

**Comment by eimfach** [comment]
Maybe you are right I was just thinking in regular programming terms.
